lib/graphql/union_type.rb

```diff
@@ -24,19 +24,21 @@ module GraphQL
   #  }
   #
   class UnionType < GraphQL::BaseType
-    accepts_definitions :possible_types
+    accepts_definitions :possible_types, :resolve_type
     ensure_defined :possible_types
 
     def initialize
       super
       @dirty_possible_types = []
       @clean_possible_types = nil
+      @resolve_type_proc = nil
     end
 
     def initialize_copy(other)
       super
       @clean_possible_types = nil
       @dirty_possible_types = other.dirty_possible_types.dup
+      @resolve_type_proc = other.resolve_type_proc
```

Comment:
I think this is the default behavior of `#initialize_copy`, to assign each ivar of the original to to the new copy, so we can remove this line and let `super` handle it.